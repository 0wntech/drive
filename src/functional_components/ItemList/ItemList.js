import React from 'react';
import { Item } from '../Item';
import styles from './ItemList.module.css';
import { File } from '../File';
import fileutils from '../../utils/fileUtils';
const ItemList = ({
    selectedItems,
    items,
    image,
    onItemClick,
    currPath,
    isFile = false,
    onDelete,
    onAccess,
    onRename,
    onInfo,
}) => {
    const itemComponents = items
        ? items.map((item, index) => {
              return isFile ? (
                  <File
                      selectedItem={
                          selectedItems.includes(
                              currPath + '/' + encodeURIComponent(item)
                          )
                              ? true
                              : undefined
                      }
                      key={item + index}
                      image={image}
                      onClick={() => {
                          console.log(currPath);
                          onItemClick(
                              currPath + '/' + encodeURIComponent(item)
                          );
                      }}
                      onDelete={onDelete}
                      onAccess={onAccess}
                      onRename={onRename}
                      onInfo={onInfo}
                      label={item}
                      currPath={currPath}
                  />
              ) : (
                  <Item
                      selectedItem={
                          selectedItems.includes(
                              currPath + encodeURIComponent(item) + '/'
                          )
                              ? true
                              : undefined
                      }
                      key={item + index}
                      image={image}
                      onClick={() =>
                          onItemClick(currPath + '/' + encodeURIComponent(item))
                      }
                      onDelete={onDelete}
                      onAccess={fileutils.changeAccess}
                      onRename={fileutils.renameFile}
                      onInfo={fileutils.getInfo}
                      currPath={currPath}
                      label={item}
                  />
              );
          })
        : undefined;

    return <div className={styles.listContainer}>{itemComponents}</div>;
};

export default ItemList;
