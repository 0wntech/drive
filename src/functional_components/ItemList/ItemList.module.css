.listContainer {
    padding: 0;
    color: blue;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(110px, 1fr));
    grid-gap: 30px 10px;
}
