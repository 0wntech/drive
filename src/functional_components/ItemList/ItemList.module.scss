@import '../../../styles/constants.scss';

.listContainer {
    color: #000;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(110px, 1fr));
    grid-gap: 30px 10px;
}

.listContainer:last-child {
    margin: 40px 0 20px 0;
}

@media (max-width: $screen_s) {
  .listContainer {
    grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
    grid-gap: 20px 20px;
  }
}
