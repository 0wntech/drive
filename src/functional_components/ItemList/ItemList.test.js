import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ItemList from './ItemList';

describe('ItemList', () => {
    const currentItem = {
        files: ['haha.jpg'],
        folders: ['hehe', 'hihi'],
    };

    describe('ItemList Snapshot', () => {
        it('matches the snapshot for items', () => {
            const result = new ShallowRenderer().render(
                <ItemList.WrappedComponent
                    items={currentItem.files}
                    selectedItems={[]}
                    currentPath="https://example.com/"
                />
            );
            expect(result).toMatchSnapshot();
        });
        it('matches the snapshot for folders', () => {
            const result = new ShallowRenderer().render(
                <ItemList.WrappedComponent
                    items={currentItem.folders}
                    selectedItems={[]}
                    currentPath="https://example.com/"
                />
            );
            expect(result).toMatchSnapshot();
        });
        it('matches the snapshot for selected folders', () => {
            const result = new ShallowRenderer().render(
                <ItemList.WrappedComponent
                    items={currentItem.folders}
                    selectedItems={currentItem.folders}
                    currentPath="https://example.com/"
                />
            );
            expect(result).toMatchSnapshot();
        });
    });
});
