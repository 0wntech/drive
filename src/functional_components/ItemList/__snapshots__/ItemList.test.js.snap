// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList ItemList Snapshot matches the snapshot for folders 1`] = `
<div
  className="listContainer"
>
  <MyItem
    contextMenuOptions={
      Array [
        Object {
          "disabled": false,
          "label": "Info",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Copy*",
          "onClick": [Function],
        },
        Object {
          "disabled": undefined,
          "label": "Paste*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Rename",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Manage Access",
          "onClick": [Function],
        },
        Object {
          "disabled": true,
          "label": "Share*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Delete",
          "onClick": [Function],
        },
      ]
    }
    currentPath="https://example.com/"
    label="hehe"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
  <MyItem
    contextMenuOptions={
      Array [
        Object {
          "disabled": false,
          "label": "Info",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Copy*",
          "onClick": [Function],
        },
        Object {
          "disabled": undefined,
          "label": "Paste*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Rename",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Manage Access",
          "onClick": [Function],
        },
        Object {
          "disabled": true,
          "label": "Share*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Delete",
          "onClick": [Function],
        },
      ]
    }
    currentPath="https://example.com/"
    label="hihi"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
</div>
`;

exports[`ItemList ItemList Snapshot matches the snapshot for items 1`] = `
<div
  className="listContainer"
>
  <MyItem
    contextMenuOptions={
      Array [
        Object {
          "disabled": false,
          "label": "Info",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Copy*",
          "onClick": [Function],
        },
        Object {
          "disabled": undefined,
          "label": "Paste*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Rename",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Manage Access",
          "onClick": [Function],
        },
        Object {
          "disabled": true,
          "label": "Share*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Delete",
          "onClick": [Function],
        },
      ]
    }
    currentPath="https://example.com/"
    label="haha.jpg"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
</div>
`;

exports[`ItemList ItemList Snapshot matches the snapshot for selected folders 1`] = `
<div
  className="listContainer"
>
  <MyItem
    contextMenuOptions={
      Array [
        Object {
          "disabled": false,
          "label": "Info",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Copy*",
          "onClick": [Function],
        },
        Object {
          "disabled": undefined,
          "label": "Paste*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Rename",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Manage Access",
          "onClick": [Function],
        },
        Object {
          "disabled": true,
          "label": "Share*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Delete",
          "onClick": [Function],
        },
      ]
    }
    currentPath="https://example.com/"
    label="hehe"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
  <MyItem
    contextMenuOptions={
      Array [
        Object {
          "disabled": false,
          "label": "Info",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Copy*",
          "onClick": [Function],
        },
        Object {
          "disabled": undefined,
          "label": "Paste*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Rename",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Manage Access",
          "onClick": [Function],
        },
        Object {
          "disabled": true,
          "label": "Share*",
          "onClick": [Function],
        },
        Object {
          "disabled": false,
          "label": "Delete",
          "onClick": [Function],
        },
      ]
    }
    currentPath="https://example.com/"
    label="hihi"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  />
</div>
`;
